yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -354,17 +475,15 @@ func (p *ParserIssueDetector) finalizeColumnsFromParentMap(parentMap map[string]
 			continue // This is a root table with no parents
 		}
 		for _, parent := range parentList {
-			parentCols, ok := p.columnMetadata[parent]
+			parentTM, ok := p.tableMetadata[parent]
 			if !ok {
 				continue // Parent metadata not found
 			}
-			if _, exists := p.columnMetadata[child]; !exists {
-				p.columnMetadata[child] = make(map[string]*ColumnMetadata)
-			}
-			for colName, colMeta := range parentCols {
-				if _, exists := p.columnMetadata[child][colName]; !exists {
+			childTM := p.getOrCreateTableMetadata(child)
+			for colName, colMeta := range parentTM.Columns {
+				if _, exists := childTM.Columns[colName]; !exists {
 					copy := *colMeta
```

Comment:
just curious: why not just directly do `childTM.Columns[colName] = colMeta` ? would that not work?